OneOneOne/OneOneOne/message_handler.py

```diff
@@ -0,0 +1,190 @@
+from lxml import etree, objectify
+import xml.etree.ElementTree as ET
+
+from builder.pystache_message_builder import PystacheMessageBuilder
+from utilities.file_utilities import FileUtilities
+from definitions import XML_PATH, TEMPLATE_PATH
+import logging
+
+basic_success_response = FileUtilities.get_file_string(XML_PATH / 'basic_success_response.xml')
+
+base_fault_response = FileUtilities.get_file_string(XML_PATH / 'basic_fault_response.xml')
+
+
+def build_error_message(error):
+    builder = PystacheMessageBuilder(str(TEMPLATE_PATH), 'base_error_template')
+    return builder.build_message({"errorMessage": error})
+
+
+class MessageHandler:
+    namespaces = {
+        'soap': 'http://schemas.xmlsoap.org/soap/envelope/',
+        'a': 'http://www.etis.fskab.se/v1.0/ETISws',
+        'wsa': 'http://www.w3.org/2005/08/addressing',
+        'itk': 'urn:nhs-itk:ns:201005'
+    }
+
+    def __init__(self, message_string):
+        logging.basicConfig(level=logging.DEBUG)
+        self.message = message_string
+        self.message_tree = ET.fromstring(self.message)
+
+        self.check_list = [
```

Comment:
I like that this is built to make it easier to add new check methods, but it might be more flexible if the list was of objects that all implement a `check` method (for example). That way a new check could be added by implementing a new class, rather than adding a method to this one. If you combined that with passing the list of checks as an argument to this constructor, this class then wouldn't need to know any details of the actual checks being run, that wiring information could live in the main application class.

Reply:
This has been updated as suggested, I now have an interface, `Check`, with generic methods and a check method, the message handler is passed a list of objects which implement the `Check` class and calls each check method